riposte-core/src/test/java/com/nike/riposte/util/AsyncNettyHelperTest.java

```diff
@@ -857,4 +866,84 @@ public void executeOnlyIfChannelIsActive_sets_original_tracing_and_mdc_info_back
         );
         assertThat(postCallTracingInfo).isEqualTo(preCallTracingInfo);
     }
+
+    @Test
+    public void executeAsyncCall_shouldReturnCompletableFuture() throws Exception {
+        // given
+        String expectedResult = UUID.randomUUID().toString();
+        ctxMock = TestUtil.mockChannelHandlerContextWithTraceInfo();
+
+        // when
+        CompletableFuture<String> completableFuture = AsyncNettyHelper.executeAsyncCall(
+                () -> expectedResult,
+                executor, ctxMock);
+
+        // then
+        assertThat(completableFuture.isCompletedExceptionally()).isFalse();
+        assertThat(completableFuture.get()).isEqualTo(expectedResult);
+    }
+
+    @Test
+    public void executeAsyncCall_shouldReturnCompletableFutureUsingCircuitBreaker() throws Exception {
+        // given
+        String expectedResult = UUID.randomUUID().toString();
```

Comment:
In the tests where a circuit breaker is involved you should verify that the circuit breaker is called as you expect. You can use a `spy(...)` for that purpose and then `verify(...)` the method calls on the circuit breaker.

Reply:
Good call. i'll add those.